openquake/commonlib/readinput.py

```diff
@@ -1066,6 +1066,20 @@ def get_exposure(oqparam, h5=None):
     return exposure
 
 
+def read_df(fname, lon, lat, id):
+    """
+    Read a DataFrame containing lon-lat-id fields and raise an error
+    for duplicate sites, if any
+    """
+    dframe = pandas.read_csv(fname)
+    dframe[lon] = numpy.round(dframe[lon].to_numpy(), 5)
+    dframe[lat] = numpy.round(dframe[lat].to_numpy(), 5)
+    for key, df in dframe.groupby([lon, lat]):
```

Comment:
Could use `dframe.duplicated(subset=[lon, lat], keep='first')`

Reply:
I did not know that trick ;-)